### ISSUE TITLE
'Class 'Braintree_Modification' not found'

### ISSUE THREAD

**Issue by schilakamarri** [braintree/braintree_php#64]
Environment: Laravel PHP and "braintree/braintree_php" : "2.35.2"
When performing a call:

             $addOns = Braintree_AddOn::all();

I get an exception:
[2014-12-18 23:14:46] production.ERROR: exception 'Symfony\Component\Debug\Exception\FatalErrorException' with message 'Class 'Braintree_Modification' not found' in /var/www/vendor/braintree/braintree_php/lib/Braintree/Discount.php:3
Stack trace:
#0 [internal function]: Illuminate\Exception\Handler->handleShutdown()
#1 {main} [] []

How do I fix this?

**Comment by jdahern** [comment]
I get the same error when I changed to 2.35.0 / 2.35.1 / 2.35.2 / 2.36.0 
2.34.0 does not have that error.

Im also using composer, and had to update the from 2.25.0


Im ok with 2.34.0 for now, but still wanted to give any info that might help out with the issue

My stack trace is:
# Fatal error: Class 'Braintree_Modification' not found in /path/to/repo/vendor/braintree/braintree_php/lib/Braintree/AddOn.php on line 3
Call Stack
n	Time		Function	Location
1	0.0009		{main}( )	.../repo.php:0
2	0.5324		ExtLoader::_loadExtension( )	.../repo.php:137
3	0.5334		invokeArgs ( )	.../extLoader.class.php:335
4	0.5334		CronSection->refreshPlanCache( )	.../extLoader.class.php:335
5	0.5341		BraintreeCacheHelper::cacheAll( )	.../cron_section.class.php:164
6	0.5344		BraintreeCacheHelper::cache_discounts( )	.../BraintreeCacheHelper.php:134
7	0.5344		spl_autoload_call ( )	.../BraintreeCacheHelper.php:117
8	0.5344		Composer\Autoload\ClassLoader->loadClass( )	.../BraintreeCacheHelper.php:117
9	0.5344		Composer\Autoload\includeFile( )	.../ClassLoader.php:278
10	0.5345		include( '/path/to/repo/vendor/braintree/braintree_php/lib/Braintree/Discount.php' )	.../ClassLoader.php:386
11	0.5345		spl_autoload_call ( )	.../ClassLoader.php:3
12	0.5345		Composer\Autoload\ClassLoader->loadClass( )	.../ClassLoader.php:3
13	0.5345		Composer\Autoload\includeFile( )	.../ClassLoader.php:278
14	0.5346		include( '/path/to/repo/vendor/braintree/braintree_php/lib/Braintree/Modification.php' )	.../ClassLoader.php:386
15	0.5346		spl_autoload_call ( )	.../ClassLoader.php:3
16	0.5346		Composer\Autoload\ClassLoader->loadClass( )	.../ClassLoader.php:3
17	0.5346		Composer\Autoload\includeFile( )	.../ClassLoader.php:278
18	0.5348		include( '/path/to/repo/vendor/braintree/braintree_php/lib/Braintree.php' )	.../ClassLoader.php:386
19	0.5354		require_once( '/path/to/repo/vendor/braintree/braintree_php/lib/Braintree/AddOn.php' )	.../Braintree.php:73

hope this helps

Jeff

**Comment by mopo922** [comment]
I also have this issue with version 2.39.0.

Well, actually mine is the opposite - I'm calling `Braintree_Discount::all()` and getting this error:

```
'Class 'Braintree_Modification' not found' in /var/www/vendor/braintree/braintree_php/lib/Braintree/AddOn.php:3
```

I can confirm that downgrading to 2.34.0 fixes the problem.

**Comment by matthewpatterson** [comment]
Hey @schilakamarri, @jdahern, and @mopo922, we've identified the problem here and are working on a fix. Thanks again for letting us know about this!

**Comment by mopo922** [comment]
Great news, @matthewpatterson . Please let us know when the fix is available.

**Comment by matthewpatterson** [comment]
This has been fixed in [version 2.40.0](https://packagist.org/packages/braintree/braintree_php#2.40.0). It is currently available through Packagist and should be updated on https://developers.braintreepayments.com soon. Thanks again for reporting this!

**Comment by mopo922** [comment]
You da man, @matthewpatterson ! Thanks for the update. I plan to upgrade later tonight.
